### PR TITLE
Plugin Messages API (Untested)

### PR DESCRIPTION
**Pull request by xxDark**

### CODE REVIEW

File: api/src/main/java/com/velocitypowered/api/proxy/ProxyServer.java [hunk 1]
```diff
@@ -19,7 +20,22 @@
 /**
  * Provides an interface to a Minecraft server proxy.
  */
-public interface ProxyServer {
+public abstract class ProxyServer {
```
Comment (md678685) [inline api/src/main/java/com/velocitypowered/api/proxy/ProxyServer.java]: Uh, what? You should be able to pass a `ProxyServer` instance wherever you should need it. This doesn't need to be turned into a singleton.

File: api/src/main/java/com/velocitypowered/api/proxy/ProxyServer.java [hunk 2]
```diff
@@ -19,7 +20,26 @@
...
+
+  private static ProxyServer instance;
```
Comment (Leymooo) [inline api/src/main/java/com/velocitypowered/api/proxy/ProxyServer.java]: We dont want to use static access to ProxyServer, just pass a instance to VelocityChannelIdentifier.

File: proxy/src/main/java/com/velocitypowered/proxy/messages/VelocityChannelIdentifier.java [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+
+/**
+ * A dummy {@link ChannelIdentifier} for communication w/ Velocity
+ */
+public final class VelocityChannelIdentifier implements ChannelIdentifier {
+
+    private VelocityChannelIdentifier() { }
+
+    private static final byte[] API_MISMATCH = new byte[]{VelocityReplyCodes.API_MISMATCH};
+    public static final ChannelIdentifier INSTANCE = new VelocityChannelIdentifier();
```
Comment (Leymooo) [inline proxy/src/main/java/com/velocitypowered/proxy/messages/VelocityChannelIdentifier.java]: I think its better to create a getter in VelocityServer.

File: api/src/main/java/com/velocitypowered/api/proxy/ProxyServer.java (hunk 1 above)
Comment (xxDark) [reply api/src/main/java/com/velocitypowered/api/proxy/ProxyServer.java]: > 
> 
> Uh, what? You should be able to pass a `ProxyServer` instance wherever you should need it. This doesn't need to be turned into a singleton.

Gonna fix it, when will deal with code style issues :(

**Comment by xxDark** [comment]
P.S Minecraft limits the amount of data being sent in payload packet, that's why I decided to comment the code where I write all player uuids/names.
To bypass this 'restriction (?)', I can make a 'trick' that splits a data in multiple payload packets. 🤔

**Comment by astei** [comment]
@xxDark It wouldn't be hard to just split up the plugin message packets, no? The 32KiB limit is pretty annoying, though.

**Comment by astei** [comment]
Note: we'll probably also need to produce a portable (Bukkit/Sponge) API for this once we're done.

**Comment by astei** [comment]
@xxDark still interested in working on this? I might be willing to take over if you've lost interest.

**Comment by xxDark** [comment]
Yup, I'm a bit busy rn.

**Pull request closed without merging**